sql/core/src/main/scala/org/apache/spark/sql/execution/SparkStrategies.scala

```diff
@@ -82,7 +82,10 @@ private[sql] abstract class SparkStrategies extends QueryPlanner[SparkPlan] {
    */
   object CanBroadcast {
     def unapply(plan: LogicalPlan): Option[LogicalPlan] = {
-      if (plan.statistics.sizeInBytes <= conf.autoBroadcastJoinThreshold) {
+      // We can't estimate the size of ObjectType
+      if (plan.find(_.isInstanceOf[ObjectProducer]).isDefined) {
```

Comment:
`ObjectProducer` does not always produce objects, think about int encoder. We should check if the output is `ObjectType`